src/main/java/org/openbase/bco/bcozy/controller/UserSettingsController.java

```diff
@@ -0,0 +1,390 @@
+                .getBundle(Constants.LANGUAGE_RESOURCE_BUNDLE, Locale.getDefault());
+
+        //TODO: Implement Property implementation
+
+        availableLanguages = FXCollections.observableArrayList("English", "Deutsch");
+        languageChoice.setItems(availableLanguages);
+
+
+        availableThemes = FXCollections.observableArrayList(
+                languageBundle.getString(Constants.LIGHT_THEME_CSS_NAME),
+                languageBundle.getString(Constants.DARK_THEME_CSS_NAME));
+        themeChoice.setItems(availableThemes);
+
+        savePassword.setApplyOnNewText(String::toUpperCase);
+        changePasswordPane.setExpanded(false);
+
+
+    }
+
+    private void updateOccupant(Boolean isOccupant) {
+        try {
+            UserConfigType.UserConfig userConfig = Registries.getUserRegistry()
+                    .getUserConfigById(SessionManager.getInstance().getUserId()).getUserConfig().toBuilder()
+                    .setOccupant(isOccupant).build();
+
+            UnitConfigType.UnitConfig unitConfig = Registries.getUserRegistry()
+                    .getUserConfigById(SessionManager.getInstance().getUserId()).toBuilder()
+                    .setUserConfig(userConfig).build();
+
+            Future<UnitConfigType.UnitConfig> saved = Registries.getUserRegistry().updateUserConfig(unitConfig);
+
+            boolean savedValue = saved.get(5, TimeUnit.SECONDS).getUserConfig().getOccupant();
+
+            if (savedValue == isOccupant) {
+                showSuccessMessage();
+
+                return;
+            }
+
+        } catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
+        } catch (CouldNotPerformException | ExecutionException | TimeoutException ex) {
+            ex.printStackTrace();
+        }
+
+        showErrorMessage();
+    }
+
+    private void onLogin() throws InterruptedException {
+        LOGGER.warn("UserID is " + SessionManager.getInstance().getUserId());
+
+        initUserName();
+        initFirstName();
+        initLastName();
+        initMail();
+        initPhone();
+        initOccupant();
+
+    }
+
+    private void initOccupant() throws InterruptedException {
+        try {
+            isOccupantField.setSelected(Registries.getUserRegistry().getUserConfigById(SessionManager.getInstance()
+                    .getUserId()).getUserConfig().getOccupant());
+
+            isOccupantField.selectedProperty().addListener((observable, oldValue, newValue) -> {
+                updateOccupant(newValue);
+
+            });
+        } catch (CouldNotPerformException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private void initPhone() throws InterruptedException {
+        try {
+            changePhone.setText(Registries.getUserRegistry().getUserConfigById(SessionManager.getInstance().getUserId
+                    ()).getUserConfig().getMobilePhoneNumber());
+        } catch (CouldNotPerformException e) {
+            e.printStackTrace();
```

Comment:
Please to not mess up the code by printing Stacktraces. Please use the ExceptionPrinter. Use find and replace for the hole bcozy project to replace this lines.

Reply:
fixed